tasks/UnderwaterTask.hpp

```diff
@@ -7,6 +7,7 @@
 #include <gazebo/physics/physics.hh>
 #include <gazebo/transport/transport.hh>
 #include <gazebo/msgs/vector3d.pb.h>
+#include <regex>
```

Comment:
Add includes in the most restrictive scope.

You use regex_replace in the `.cpp`, add the include there.